### PR TITLE
Add protection domain

### PR DESCRIPTION
**Pull request by JacobGros**
# Description
Allows driver to find storage pool based on Protection Domain and name; rather than just name. 
# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/415 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Added and ran unit tests
- [x] Ran int tests
- [x] Ran E2E tests

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,7 +3,7 @@ module github.com/dell/csi-vxflexos/v2
 // In order to run unit tests on Windows, you need a stubbed Windows implementation
 // of the gofsutil package. Use the following replace statements if necessary.
 
-//replace github.com/dell/gofsutil => ./gofsutil
+//replace github.com/dell/goscaleio => ./goscaleio
```
Comment (bharathsreekanth) [inline go.mod:6]: If we are not using this, do we even need the commented lines?

File: samples/storageclass/storageclass.yaml [hunk 2]
```diff
@@ -24,6 +24,10 @@ parameters:
   # Storage pool to use on system
   # Optional: false
   storagepool: <STORAGE_POOL> # Insert Storage pool
+  # Protection domain that storage pool above belongs to
+  # Needed if array has two storagepools that share the same name, but belong to different domains
```
Comment (bharathsreekanth) [inline samples/storageclass/storageclass.yaml:28]: minor : but belong to different protection domains.

File: service/controller.go [hunk 3]
```diff
@@ -26,6 +26,10 @@ const (
 	// volume create parameters map
 	KeyStoragePool = "storagepool"
 
+	// KeyProtectionDomain is the key used to get the StoragePool's Protection Domain name from the
+	// volume create parameters map. This pramater is optional.
```
Comment (bharathsreekanth) [inline service/controller.go:30]: typo in parameter.

**Review by bharathsreekanth — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (JacobGros) [reply go.mod]: Given the switch to internal tags, we do not need them anymore. Removed them.

File: samples/storageclass/storageclass.yaml (hunk 2 above)
Comment (JacobGros) [reply samples/storageclass/storageclass.yaml]: fixed

File: service/controller.go (hunk 3 above)
Comment (JacobGros) [reply service/controller.go]: fixed

**Review by jooseppi-luna — APPROVED** [review]

**Review by bharathsreekanth — APPROVED** [review]

**Pull request merged**